### DISCUSSION TITLE
Professional Firmware Configurator

### DISCUSSION THREAD

**Discussion by I-wish-2** [mriscoc/Ender3V2S1#638] [Q&A]
Greetings,
I am making an attempt to use the Configurator tool and I'm not sure of how to get it up and running.  I have installed Python 3.1 and have copied the lines of code from the configurator.puw file in Special Configurations.
Now being nubee I'm not sure how to get the configurator to wake up in Pylon.  If anyone might be able to help me get going I would be most thankful.  
Doug

**Comment by mriscoc** [comment]
In the https://github.com/mriscoc/Special_Configurations repository there is a [readme](https://github.com/mriscoc/Special_Configurations/blob/main/README.md) file with instructions.

**Comment by I-wish-2** [comment]
Thank you I’m on it.

D

**Reply by I-wish-2** [reply]
Greetings,
Thank you Miguel for the information and directions.  After a some effort I was able to compile an Experimental bin file from all of the previous work you did for me.  I made some notes along the way and thought that they might help someone new who is to trying to produce a basic or experimental bin file.

Well here is what I did and please feel free to RED line it where it need clarification.

                         Experimental Printer Configuration and Compiling a new bin file.

1.	From the Special_Configurations Repository, https://github.com/mriscoc/Special_Configurations,  click Latest (Right Side ) and download the Source Code (Zip).   In addition review the readme file for examples.
2.	Un-Zip all the files to a new Folder on your system.    My Folder Example, "Special_Configurations-InputShaping"
3.	Open the Folder and scroll to the bottom and look for the Configurator.pyw file'  
4.	Double Click on the file to start it. Note: Python3.XX must be installed on your computer.
  	Select your Printer, M. Board, Leveling, Thermistor ( Enders use the T1 ) and Features like MPC for sure and maybe LA.
5.	Click " Set config ", then click the upper right " Auto "  this will place a configuration file name in the window.
6.	Edit the file name as you need it to be now click " Generate " this will save the configuration file, you named, back in the main folder.
7.	Look in the main folder where the Configurator.pyw file is and find the file you saved from the Configurator.
8.	Copy the Configuration.h & Configuration.adv.h files from it, and maybe paste them on the desktop. For step 11 below.


9.	Now from the Ender3V2S1 Repository,  https://github.com/mriscoc/Ender3V2S1,  click Latest (Right Side ) scroll to the bottom and download the Source Code (Zip).   Note this is a different Source Code than from the file above in step 1.
10.	Un-Zip all the files to a different new Folder on your system. A  different one the one from step 2.    My Folder Example, "Ender3V22S1-20221222"
11.	From step 8 above, Copy the Configuration.h & Configuration.adv.h files to the Marlin folder in the new un-zip folder you created and over write the originals.


12.	Open Visual Studio Code.  Select open folder and Open the complete folder from step 10 with the updated Configuration.h & Configuration.adv.h files.

13.	Test the compile by clicking the Bottom bar check mark ( PlatformIO: Build )
14.	Make any additional edits you require and re-compile.
15.	Your bin file is located at the top of the left hand pane,  click:>  .pio\build > STM32F103RC_creality.  Drag and drop the top listed firmware-xxxxxxxx-xxxxxx.bin to your desk top.
16.	Prior to flashing your mother board make a copy of all the current configuration and calibration data on your printer, you will need it.
17.	Copy the bin file to a formatted 8gig SD card, install in your printer and power up.  Re-enter your printers old configuration data.  After entering all of your previous configuration data, save it all, then Power Off your printer and remove the SD card.  Then power back on and do all the auto temperature calibrations, tramming, z-offsets and bed mesh.  Save it all again.  Power off one more time prior to running a print.  This will make sure all the data you saved is in memory correctly.

18.	Bob's your Uncle not your Aunty Jane.  Beverage time, Cheers.


I do feel way more confident now and I hope this effort helps out.
Doug

**Reply by I-wish-2** [reply]
Greetings,
There are some additional RED line edits to above steps to include mother boards other than Creality 4.2.2 / 4.2.7. as well as a Marlin Firmware Auto Build extension that includes additional environments supported by the boards.

If you see any additional edits for clarification please add them.

Changes were made in Steps 8,  11,  original step 13 & 18 deleted,  New steps 13 thru 17 edited or re-numbered.

Here is the revision:

  Experimental Printer Configuration and Compiling a new bin file revision 1

1.   From the Special_Configurations Repository, https://github.com/mriscoc/Special_Configurations, click Latest (Right Side ) and download the Source Code (Zip). In addition review the readme file for examples.

2.   Un-Zip all the files to a new Folder on your system. My Folder Example, "Special_Configurations-InputShaping"

3.   Open the Folder and scroll to the bottom and look for the Configurator.pyw file'

4.   Double Click on the file to start it. Note: Python3.XX must be installed on your computer.
Select your Printer, M. Board, Leveling, Thermistor ( Enders use the T1 ) and Features like MPC for sure and maybe LA.

5.   Click " Set config ", then click the upper right " Auto " this will place a configuration file name in the window.

6.   Edit the file name as you need it to be now click " Generate " this will save the configuration file, you named, back in the main folder.

7.  Look in the main folder where the Configurator.pyw file is and find the file you saved from the Configurator.

8.   Copy the Configuration.h, Configuration.adv.h & Platformio.inf files from it, and maybe paste them on the desktop. For step 11 below.

9.   Now from the Ender3V2S1 Repository, https://github.com/mriscoc/Ender3V2S1, click Latest (Right Side ) scroll to the bottom and download the Source Code (Zip). Note this is a different Source Code than from the file above in step 1.

10.  Un-Zip all the files to a different new Folder on your system. A different one the one from step 2. My Folder Example, "Ender3V22S1-20221222"

11.  From step 8 above, Copy the Configuration.h & Configuration.adv.h files to the Marlin folder in the new un-zip folder you created and over write the originals.   Copy the Platformio.inf file to the Main folder where the Marlin folder is and overwrite the current one.

12.  Open Visual Studio Code. Select open folder and Open the complete folder from step 10 with the updated Configuration.h & Configuration.adv.h and Platformio.inf files.
 
13.  After editing, Test the compile by clicking the M icon, which is the " Auto Build Marlin " on the far left tool bar.  Then click the Show ABM Panel and scroll down to Environments and select the Build for your Environment.  The chip on my 4.2.7 board is a STM32f103RET6 (512k).  The first listing was for a STM32F103RE_Creality (512K) so I selected it and clicked Build.

14.  When the compile is complete your build selection will turn GREEN and it will transfer the compiled bin and elf file to the hard drive.  Look in the source code folder, the one you loaded into Visual Studio Code, open the pio folder  >  build folder  >  ( in my case the STM32F103RC_creality ) folder and the the bin is listed there.  You could also use step 15.

15.  Your bin file is located at the top of the left hand pane, click:> .pio\build > STM32F103RC_creality. Drag and drop the top listed firmware-xxxxxxxx-xxxxxx.bin to your desk top.

16.  Prior to flashing your mother board make a copy of all the current configuration and calibration data on your printer, you will need it.

17.  Copy the bin file to a formatted 8gig SD card, install in your printer and power up. Re-enter your printers old configuration data. After entering all of your previous configuration data, save it all, then Power Off your printer and remove the SD card. Then power back on and do all the auto temperature calibrations, tramming, z-offsets and bed mesh. Save it all again. Power off one more time prior to running a print. This will make sure all the data you saved is in memory correctly.

Thank you for reviewing.
Doug

**Comment by I-wish-2** [comment]
Hi Miguel,
In the Marlin Firmware Auto Build screen in Visual Studio Code how do you determine which environmental listing to select to build the compile from.
As always Thank You.
Doug

**Reply by mriscoc** [reply]
The environment depends on the SoC that your mainboard has, `STM32F103RE_creality` is for RET6 devices, `STM32F103RC_creality` is for RCT6 devices. The `*_xfer` environments are for special flash transferring from PC using BFT.

**Comment by I-wish-2** [comment]
Thank you Miguel for the clarification.  I have a few more results from the procedure steps for posting soon.

Best Regards

Doug

**Reply by mriscoc** [reply]
When you're done, we can clean this post up a bit and turn it into a guide.

**Comment by I-wish-2** [comment]
Hi Miguel,

I would be honored if the post became a guide.  I am having one issue that I have not resolved as of yet.  When I copy the Platformio.ini file from the Professional Firmware Configurator, ( ReadME.md file “move the platform.io file to the root of your project folder.” ) and then build the compile works great.  When I save the newly created bin file to the SD card and Install it in the printer and power it up the printer just beeps and beeps and beeps and will not flash.

If I do not copy the platform.io file to the root of my project folder, the build, compile and flash works great and the printer functions with the new bin file edits fine.  Am I missing something?  I’m am going to send some more time today to try to narrow down where I missing something.  Updates to follow.

 

My Best Regards

Doug

**Comment by I-wish-2** [comment]
Greetings,
I resolved the above platformIO.inf issue as a defective SD card.  Static sensitive is suppose.  I can successfully compile bin files using the updated steps below.  Please let me know about the clean up efforts and if I can help.
I do have one additional question relating to the Environments that you select to Build bin file from.

In Visual Studio Code > (M) Auto Build Marlin > Show ABM Panel > I have a listing of 3 Environments for my 4.2.7 board with the STM32F103RE chip.

1.  STM32F103RE_creality (512k)
2. STM32F103RE_creality_xfer (512k)
3. STM32F103RE_creality_maple (512k)

For future reference could you please explain the differences and best choice.

Here are the current guide steps.



                                              Experimental Configurations and Compiling a new bin file.

Section 1.

1.   From the Special_Configurations Repository, https://github.com/mriscoc/Special_Configurations, click Latest (Right Side ) and download the Source Code (Zip). In addition review the readme file for how to examples.  This readme is in the main repository.

2.   Un-Zip all the files to a new Folder on your system. My Folder Example, "Special_Configurations-InputShaping"

3.   Open the Folder and scroll to the bottom and look for the Configurator.pyw file'

4.   Double Click on the file to start it. Note: Python3.XX must be installed on your computer first.
Select your Printer, M. Board, Leveling, Thermistor ( Enders use the T1 ) and Features like MPC for sure and maybe LA, Linear Advance.

5.   Click " Set config ", then click the upper right " Auto " this will place the configuration file name of your selections in the window.

6.   Edit the file name as you required and then click " Generate ",  this will save the configuration file, you named, back in your main folder.

7.  Look in the main folder where the Configurator.pyw file is and find the file you saved from the Configurator and double click it.

8.   Copy the Configuration.h, Configuration.adv.h & Platformio.inf files from it, and maybe paste them on the desktop. For step 11 below.


Section 2.

9.   Now from the Ender3V2S1 Repository, https://github.com/mriscoc/Ender3V2S1, click Latest (Right Side ) scroll to the bottom and download the Source Code (Zip). Note this is a different Source Code than from the file above in Section 1 step 1.

10.  Un-Zip all the files to a different new Folder on your system. A different one from the one in Section 1 step 2. My Folder Example, "Ender3V22S1-20221222"

11.  From step 8 above, Copy the Configuration.h & Configuration.adv.h files to the Marlin folder in the new un-zip folder you created and over write the originals.   Copy the Platformio.inf file to the Main folder where the Marlin folder is and overwrite the current one.

12.  Open Visual Studio Code. Select Open Folder and open the complete folder from step 10 with the updated Configuration.h & Configuration.adv.h and Platformio.inf files.
 
13.  After making any special edits, Test the compile by clicking the M icon, which is the " Auto Build Marlin " on the far left tool bar.  Then click the Show ABM Panel and scroll down to Environments and select the Build for your Environment.  The chip on my 4.2.7 board is a STM32f103RET6 (512k).  The first listing was for a STM32F103RE_Creality (512K) so I selected it and clicked Build.

14.  When the compile is complete your build selection will turn GREEN and it will transfer the compiled bin and elf file to the hard drive.  Look in the source code folder, the one you loaded into Visual Studio Code, open the pio folder  >  build folder  >  ( in my case the STM32F103RC_creality ) folder and the the bin is listed there.  You could also use step 15.

15.  Your bin file is located at the top of the left hand pane, click:> .pio\build > STM32F103RC_creality. Drag and drop the top listed firmware-xxxxxxxx-xxxxxx.bin to your desk top.

16.  Prior to flashing your mother board make a copy of all the current configuration and calibration data on your printer, you will need it.

17.  FLASHING:  Copy the bin file to a formatted 8 / 16 gig SD card, install in your printer and power up. When the flash and " please wait until reboot " is complete.  Re-enter your printers old configuration data. After entering all of your previous configuration data, save it all, then Power Off your printer and remove the SD card. Then power back on and do all the auto temperature calibrations, tramming, z-offsets and bed mesh. Save it all again. Power off one more time prior to running a print. This will make sure all the data you saved is in memory correctly.

18.  If during the Flashing process the printer just beeps and will not Flash.  Try replacing the SD card.  They are static prone and SD cards greater that 16 gig will at times have issues if they are used for flashing new bin files.

Doug

**Reply by mriscoc** [reply]
* STM32F103RE_creality (512k)
That is for RET6 SoC

* STM32F103RE_creality_xfer (512k)
RET6 SoC and enable firmware transfer using USB in VSCode for developers

* STM32F103RE_creality_maple (512k)
RET6 SoC with deprecated maple STM library, used for legacy compatibility

**Comment by I-wish-2** [comment]
Greetings,

 

Thank you for the information as that clarifies the entries.  Is it safe to always pick the top listing?  I think the guide layout is complete in its base form let me know about its cleanup if you need me to help with it.

 

Best Regards

Doug

**Reply by I-wish-2** [reply]
Hi Miguel, sorry for the delay as I have been working hard to work out a Solvo laser upgrade to one of my Ender3 Pro printers.  Thank you for addressing the build listing in Visual Studio Code. 
Is there anything else I can help with regarding the Guide?
My Best Regards
Doug

**Reply by mriscoc** [reply]
Hi Doug, I will review this later, I have little time these days.

Regards.

**Reply by I-wish-2** [reply]
Greetings Miguel,  I do understand.. if I can help I'm here.
Best Regards
Doug

**Comment by fuzelet** [comment]
I created a Ender6 printer file and a v431 board file if anyone is looking for them.  Made some slight personal alterations to be in line with the S1 Pro F4 variant as that is another printer I use and wanted them to be feature parity("comment": "Ender6 Config parity with S1 Pro").  Not sure if it will help anyone but wanted to provide them here. Let me know if this isn't the right place, etc.


[v431_boards.zip](https://github.com/mriscoc/Ender3V2S1/files/10745460/v431_boards.zip)
[ender6_printers.zip](https://github.com/mriscoc/Ender3V2S1/files/10745461/ender6_printers.zip)

**Reply by mriscoc** [reply]
Sadly, Creality also has another board labeled as 4.3.1 but for some Ender 3V2, that board is actually a 4.2.2 board. Due to that, I can't put a config board version 4.3.1 to avoid confusion.

**Reply by fuzelet** [reply]
Appreciate the response and good information to know. Would it be possible to use/add a blank board option that doesn't make any changes and then incorporate the board variables into the ender6 printer config to avoid that confusion?  Just trying to help provide another option for your firmware.

**Comment by Mifftup** [comment]
I've followed all the steps in this guide and I'm not able to successfully compile a firmware in VSCode.

I keep getting the following error:


```
Processing STM32F103RC_creality (board: genericSTM32F103RC; platform: ststm32@~12.1; framework: arduino)   
-----------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option                                                     Processing Professional Firmware requirements
STM32F1 Architecture detected
Manual Mesh Bed Leveling detected
CONFIGURATION: https://docs.platformio.org/page/boards/ststm32/genericSTM32F103RC.html
PLATFORM: ST STM32 (12.1.1) > STM32F103RC (48k RAM. 256k Flash)
HARDWARE: STM32F103RCT6 72MHz, 48KB RAM, 256KB Flash
DEBUG: Current (jlink) External (blackmagic, cmsis-dap, jlink, stlink)
PACKAGES:
 - framework-arduinoststm32 @ 4.10900.200819 (1.9.0)
 - framework-cmsis @ 2.50501.200527 (5.5.1)
 - toolchain-gccarmnoneeabi @ 1.90201.191206 (9.2.1)
Converting Marlin.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 10 compatible libraries
Scanning dependencies...
Dependency Graph
|-- IWatchdog @ 1.0.0
|-- SPI @ 1.0
|-- EEPROM @ 2.0.1
|-- SoftwareSerial @ 1.0
|-- Wire @ 1.0
|-- Servo @ 1.1.2
Building in release mode
Compiling .pio\build\STM32F103RC_creality\FrameworkArduinoVariant\PeripheralPins.c.o
Compiling .pio\build\STM32F103RC_creality\FrameworkArduinoVariant\variant.cpp.o
arm-none-eabi-gcc: fatal error: no input files
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\HardwareSerial.cpp.o
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\HardwareTimer.cpp.o
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\IPAddress.cpp.o
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\Print.cpp.o
The system cannot find the path specified.
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\RingBuffer.cpp.o
The system cannot find the path specified.
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\Stream.cpp.o
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\Tone.cpp.o
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\USBSerial.cpp.o
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
arm-none-eabi-g++: fatal error: no input files
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\VirtIOSerial.cpp.o
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
*** [.pio\build\STM32F103RC_creality\FrameworkArduinoVariant\PeripheralPins.c.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduinoVariant\variant.cpp.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\HardwareSerial.cpp.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\HardwareTimer.cpp.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\IPAddress.cpp.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\Print.cpp.o] Error 1
Compiling .pio\build\STM32F103RC_creality\FrameworkArduino\WInterrupts.cpp.o
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\RingBuffer.cpp.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\Stream.cpp.o] Error 1
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\Tone.cpp.o] Error 1
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\USBSerial.cpp.o] Error 1
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\VirtIOSerial.cpp.o] Error 1
arm-none-eabi-g++: fatal error: no input files
compilation terminated.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
*** [.pio\build\STM32F103RC_creality\FrameworkArduino\WInterrupts.cpp.o] Error 1
======================================= [FAILED] Took 4.74 seconds =======================================

Environment           Status    Duration
--------------------  --------  ------------
STM32F103RC_creality  FAILED    00:00:04.741
================================== 1 failed, 0 succeeded in 00:00:04.741 ==================================
                                                                                                            *  The terminal process "C:\Users\M&K\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit 
code: 1.                                                                                                   
```

I have re-extracted all the original files again and tried to compile without making any changes, and I still get the same error.
I also get the same error whether I try to build using Platformio or if I try using Auto Build Marlin. Any ideas as to what I may be missing?

**Comment by I-wish-2** [comment]
Greetings,
I just followed the 1/19 guide listing as it is the last update.  I made new folders on my system for the  "Special_Configurations" and the "Ender3V2S1" zip files, this folder is now named Test Source Code, I unzipped them both and then followed the Guide.  Using the "Configurator.pyw" app I built a 427-BLTUBL-IS-LA-MPC configuration and then copied the two Configuration files to the Marlin folder and the platformio.ini file to the root folder where the Marlin folder resides.  I then opened the V.S. Code editor and loaded the Test Source Code folder with the new configurations and using Auto Build Marlin function and the first listed Environments " STM32F103RE_creality (512k)". It successfully compiled the new configuration.

With that said I might offer to make sure the two configuration files are placed the Marlin folder and the platformio.ini file is placed in your root folder that contains the Marlin folder.  I made that error several times and the compile would error out.
Since the only changes that are being made are with the Configuration.h , Configuration_adv.h and the platformio.ini files and they are placed in the correct folders then everything else should load and compile correctly.

I looked in my .pio > build > STM32F103RE_creality > FrameworkArduinoVariant folders but did not find a [ WInterrupts.cpp.o ] entry.

Sorry for not being more helpful.  Maybe someone else has seen this "WInterrupts.cpp.o" error before, if so please chime in.
Doug

**Reply by Mifftup** [reply]
Seems to be a problem with Platformio
https://github.com/platformio/platformio-core/issues/4270

But there is no specific cause listed just a new version released that fixed it for most users for a while but now it seems to be re-occurring since January this year for some users.
I've deleted my .platformio folder about 30 times.
I've uninstalled and reinstalled everything multiple times, and on multiple different Drives.
I've tried PIO Core version 5.2.5, version 6.0.1, version 6.1.1, version 6.1.6 and now I give up.

**Reply by I-wish-2** [reply]
I hear your frustration, as nothing is worse than I can't get there from here.  I must say here that platformio bugs are over my head for sure.
My current PlatformIO is Core 6.1.6 and Home is 3.4.3

What about an uninstall of the Visual Studio Code and its components.  If it's a Windows based system, scrub the System Registry file, CCleaner has a fair registry cleaner, then reinstall Visual Studio Code plugins.  Not much of a suggestion for sure. Sorry for not being much help.
Doug
